
<launch>

    <!-- must specify on command line r or l -->
    <arg name="arm"/>

    <!-- launch nodes for the fabric tactile sleeve -->
    <include file='$(find hrl_fabric_based_tactile_sensor)/launch/pr2_tactile_sleeve_combined.launch'>
        <arg name="arm" value="$(arg arm)"/>
    </include>

    <!-- convert taxel array msg to SkinContact -->
    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="forearm_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_fabric_forearm_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_forearm'/>
    </node>

    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="upperarm_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_fabric_upperarm_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_upperarm'/>
    </node>

    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="gripper_right_link_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_fabric_gripper_right_link_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_gripper_right_link'/>
    </node>

    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="gripper_left_link_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_fabric_gripper_left_link_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_gripper_left_link'/>
    </node>

    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="gripper_palm_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_fabric_gripper_palm_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_gripper_palm'/>
    </node>

    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="left_fingertip_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_pps_left_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_fingertip_left'/>
    </node>

    <node pkg="hrl_common_code_darpa_m3" type="taxel_array_to_skin_contact.py"
        output="screen" name="right_fingertip_taxel_array_to_skin_contact">
        <remap from='/skin/taxel_array_meka' to='/pr2_pps_right_sensor/taxels/forces'/>
        <remap from='/skin/contacts' to='/skin/contacts_fingertip_right'/>
    </node>

    <node pkg="sandbox_advait_darpa_m3" type="log_and_monitor_node.py" 
        name="log_and_monitor_node" args="--pr2 --disable --arm=$(arg arm)" >
    </node>

    <!-- pasue and resume with the joystick -->
    <node pkg="sandbox_advait_darpa_m3" type="pause_with_joystick.py" 
        name="pause_with_joystick_node" >
    </node>

</launch>


